pkg/labels/applicator.go

```diff
@@ -54,6 +54,11 @@ func (l Labeled) SameAs(o Labeled) bool {
 	return l.ID == o.ID && l.LabelType == o.LabelType
 }
 
+type WatchResult struct {
+	Labeled []Labeled
+	Valid   bool // Will be false if the underlying watcher has terminated
```

Comment:
is this necessary? can callers not use `range` or check `ok` when pulling a value off the channel?